Hi @another-rex,


The lint-plugger failure is a pre-existing upstream issue unrelated to this PR. The failing check points to:
extractor/filesystem/os/chisel/chisel.go:62:1: plugger: plugin.Plugin is not registered
The chisel extractor exists in google:main but is not registered in list.go — this was already present before my branch was created. My PR only adds os/wolfi and does not touch chisel.


All 12 other checks pass (CLA, CodeQL, lint on all 3 platforms, tests on all 3 platforms, tidy, linelint, filenames).


Please let me know if a rebase or any other changes are needed. 



Thank you!

--- a/extractor/filesystem/language/dotnet/paketdependencies/paketdependencies.go
+++ b/extractor/filesystem/language/dotnet/paketdependencies/paketdependencies.go
@@ -256,7 +256,6 @@ func (e Extractor) parseDependenciesFile(reader io.Reader, path string) ([]*extr
 			packages = append(packages, pkg)
 			continue
 		}
-
 	}
 
 	if err := scanner.Err(); err != nil {

--- a/extractor/filesystem/os/ecosystem/ecosystem.go
+++ b/extractor/filesystem/os/ecosystem/ecosystem.go
@@ -39,6 +39,15 @@ func MakeEcosystem(metadata any) osvecosystem.Parsed {
 	osVersionID := ""
 	switch m := metadata.(type) {
 	case *apkmeta.Metadata:
+		// Check for specific distros that use APK but map to different ecosystems.
+		// These checks must come before the version == "" guard because rolling
+		// distros (Wolfi, Chainguard) may not set VERSION_ID in os-release.
+		if m.OSID == "wolfi" {
+			return osvecosystem.FromEcosystem(osvconstants.EcosystemWolfi)
+		}
+		if m.OSID == "chainguard" {
+			return osvecosystem.FromEcosystem(osvconstants.EcosystemChainguard)
+		}
 		version := m.ToDistro()
 		if version == "" {
 			return osvecosystem.FromEcosystem(osvconstants.EcosystemAlpine)

--- a/extractor/filesystem/os/ecosystem/ecosystem_test.go
+++ b/extractor/filesystem/os/ecosystem/ecosystem_test.go
@@ -89,6 +89,20 @@ func TestMakeEcosystemAPK(t *testing.T) {
 			},
 			want: "BellSoft Hardened Containers:stream",
 		},
+		{
+			desc: "Wolfi",
+			metadata: &apkmeta.Metadata{
+				OSID: "wolfi",
+			},
+			want: "Wolfi",
+		},
+		{
+			desc: "Chainguard",
+			metadata: &apkmeta.Metadata{
+				OSID: "chainguard",
+			},
+			want: "Chainguard",
+		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.desc, func(t *testing.T) {

